src/PrestaShopBundle/Twig/Locator/ModuleTemplateLoader.php

```diff
@@ -38,27 +38,62 @@ class ModuleTemplateLoader extends FilesystemLoader
      */
     private $rootPath;
 
+    /**
+     * @var array $registeredPaths The list of Paths and namespaces registered in configuration.
+     * @see app/config/config.yml
+     */
+    private $registeredPaths;
+
     /**
      * @param string|array $paths    A path or an array of paths where to look for templates
      * @param string|null  $rootPath The root path common to all relative paths (null for getcwd())
-     * @param string|null  $namespace A path namespace
+     * @param array  $namespaces A collection of path namespaces with namespace names.
      */
-    public function __construct($paths = array(), $rootPath = null, $namespace = 'PrestaShop')
+    public function __construct($paths = array(), $rootPath = null, array $namespaces)
     {
         $this->rootPath = (null === $rootPath ? getcwd() : $rootPath).DIRECTORY_SEPARATOR;
         if (false !== $realPath = realpath($rootPath)) {
             $this->rootPath = $realPath.DIRECTORY_SEPARATOR;
         }
 
         if ($paths) {
+            $this->registerNamespacesFromConfig($paths, $namespaces);
+        }
+    }
+
+    /**
+     * Register namespaces in module and link them to the right paths.
+     * @param $paths
+     * @param array $namespaces
+     */
+    private function registerNamespacesFromConfig($paths, array $namespaces)
+    {
+        foreach($namespaces as $namespace => $namespacePath) {
             $templatePaths = array();
 
             foreach ($paths as $path) {
-                if (is_dir($dir = $path . '/views/'. $namespace)) {
+                if (is_dir($dir = $path . '/views/PrestaShop/' . $namespacePath)) {
                     $templatePaths[] = $dir;
                 }
             }
             $this->setPaths($templatePaths, $namespace);
         }
     }
+
+    /**
+     * @return array
```

Comment:
Why add this two functions? Were are they used?

Reply:
some tests, removed :+1: